src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,251 @@
+                { placeHolder: "Select whether to enable/disable the distributed tracing...", ignoreFocusOut: true },
+            );
+            if (!selectedItem) {
+                return;
+            }
+            mode = selectedItem.distributedTracingEnabled;
+        }
+
+        if (updateType !== DistributedSettingUpdateType.OnlyMode) {
+            if (mode !== false) {
+                samplingRate = await this.promptForSamplingRate(`Enter sampling rate, integer within [0, 100]`, samplingRate);
+
+                if (samplingRate === undefined) {
+                    return;
+                }
+            }
+        }
+
+        TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingStartEvent);
+
+        await vscode.window.withProgress({
+            title: `Update Distributed Tracing Setting`,
+            location: vscode.ProgressLocation.Notification,
+        }, async () => {
+            try {
+                const result = await this.scheduleTwinUpdate(mode, samplingRate, iotHubConnectionString, deviceIds);
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Success" }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel,
+                    `Update distributed tracing setting for device [${deviceIds.join(",")}] complete!` + result);
+            } catch (err) {
+                TelemetryClient.sendEvent(Constants.IoTHubAIUpdateDistributedSettingDoneEvent, { Result: "Fail", Message: err.message }, iotHubConnectionString);
+                this.outputLine(Constants.IoTHubDistributedTracingSettingLabel, `Failed to get or update distributed setting: ${err.message}`);
+                return;
+            }
+        });
+    }
+
+    private async scheduleTwinUpdate(enable: boolean, samplingRate: number, iotHubConnectionString: string, deviceIds: string[]) {
+        let twinPatch = {
+            etag: "*",
+            properties: {
+                desired: {},
+            },
+        };
+
+        if (enable === undefined && samplingRate === undefined) {
+            return;
+        }
+
+        if (!twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME]) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME] = {};
+        }
+
+        if (enable !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_mode = enable ? 2 : 1;
+        }
+
+        if (samplingRate !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_rate = samplingRate;
+        }
+
+        if (deviceIds.length === 1) {
+            return new Promise((resolve, reject) => {
+                let registry = iothub.Registry.fromConnectionString(iotHubConnectionString);
+
+                registry.updateTwin(deviceIds[0], JSON.stringify(twinPatch), twinPatch.etag, (err) => {
```

Comment:
the latest SDK, registry.updateTwin is promise now, no callback anymore

Reply:
fixed